Skyline.DataMiner.Sdk - Support CatalogReferences (#4357)
SkylineCommunications/dataminer-docs: develop/CICD/Skyline DataMiner Software Development Kit/skyline_dataminer_sdk_dataminer_package_project_catalog_references.md

      <Selection>
         <Specific>1.0.0.1</Specific>
      </Selection>
   </CatalogReference>
   <CatalogReference id="guid">
      <Name>Example Catalog Item (Range)</Name>
      <Selection>
         <Range>1.0.0</Range>
      </Selection>
   </CatalogReference>
   <CatalogReference id="guid">
      <Name>Example Catalog Item (Range - Prerelease)</Name>
      <Selection>
         <Range allowPrerelease="true">1.0.0</Range>
      </Selection>
   </CatalogReference>
-->
</CatalogReferences>
```

## Identifier

The identifier, the **id** attribute, is the GUID that represents a Catalog item. This can be retrieved from the URL, e.g.: [https://catalog.dataminer.services/details/4abcf220-c001-4ffd-bab8-559dee47088f](https://catalog.dataminer.services/details/4abcf220-c001-4ffd-bab8-559dee47088f) (Microsoft Platform)

## Name

The **Name** is meant for human readability. It is also used as the file name for the artifact. If this is empty or not present, the value from the *id* attribute will be used.

## Selection

### Specific

The *Selection.Specific* element represents a specific version that you can see in the Catalog UI.

### Range

> The *Selection.Range* element specifies a range to take the latest released version.

Reviewer: This isn't entirely clear. Is this what you mean?
```suggestion
The *Selection.Range* element specifies the range where the latest released version is added.
```

For more information about the range, see [Versioning of Catalog items](xref:About_the_Catalog_module#versioning-of-catalog-items).

In case you want prereleases as well, you can use the **allowPrerelease** attribute.

> A prerelease is defined as having a dash ("-") in the version. Note that this does not apply for the old versioning of install packages (A.B.C-CUx).

Reviewer: In what way does this not apply for the old versioning? Is any info available about this old versioning elsewhere?

Author: Changed a bit
